.github/workflows/publish-dts-sdk.yml

```diff
@@ -0,0 +1,45 @@
+name: Publish Durable Task Scheduler to PyPI
+
+on:
+  push:
+    tags:
+      - 'v*'  # Only run for tags starting with "v"
+
+jobs:
+  publish:
```

Comment:
before this intrdouce another job:

"test" which runs the unit tests, linter, type checker

Then make the "publish" job here depend on the "test" job via the `needs: {jobname}` directive.